frontend/src/metabase/actions/containers/ActionCreator/QueryActionEditor.tsx

```diff
@@ -9,19 +9,21 @@ import { getTemplateTagParameters } from "metabase-lib/parameters/utils/template
 
 function QueryActionEditor({
   question,
-  setQuestion,
+  onChangeQuestionParameters,
```

Comment:
This is just a name change. At first I want to change this to `onChangeActionParameters`, but as we [had discussed](https://metaboat.slack.com/archives/C04DAKFDTCZ/p1674138664899499). `question` is an implementation detail of the Actions.

But I want to make this callback more expressive. `setQuestion` doesn't tell anything much about when this callback would be called as the name is too generic. It might not be apparent from the parent component why we need to pass `setQuestion`. So I think the name change would hopefully improve the readability. Feel free to say otherwise, I'm happy to revert this change if it doesn't do it job well enough.

Reply:
I don't feel too strongly about this, but I think this could be more confusing since the argument is still a full question object, not just parameters.  It's probably a toss-up caused by the confusing situation that the query editor gives us a question object even though we're not actually using a lot of what a Question gives us.

Reply:
You have a good point. Lemme try to figure this out a bit more, maybe I could make this callback only accept new arguments.

Reply:
@iethree I lift all the logic up to the main component `<ActionCreator />`. Rename the callback `onChangeQuestionQuery` and accept a `NativeQuery` as a parameter. I think this should fix all the confusion. And you have another look at this?

Reply:
Yeah, I think that's a bit clearer with the logic up there. 👍